src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1153,7 +1153,17 @@ private HttpResponseHeaders CreateResponseHeaders(bool appCompleted)
         {
             if (!CanIncludeResponseContentLengthHeader())
             {
-                RejectInvalidHeaderForNonBodyResponse(appCompleted, HeaderNames.ContentLength);
+                if (responseHeaders.ContentLength.Value == 0)
```

Comment:
```suggestion
                if (responseHeaders.ContentLength.GetValueOrDefault() == 0)
```
Small performance nit:

Value property checks again for null. GetValueOrDefault returns the nullable value without any check.

(I think there has been discussion about optimizing this via JIT, but I don't know where it ended up)

Reply:
It looks like this has been fixed, likely by https://github.com/dotnet/runtime/pull/43811

I see identical asm with 6.0 between the two:

https://sharplab.io/#v2:C4LghgzgtgPgAgJgIwFgBQcAMACOSB0ASgK4B2wAllAKb4DCA9lAA4UA21ATgMpcBuFAMbUIAbnTpEuJAHZ0Ab3TZl0gGy4ALNgCyYCqQAUASmzzsABQYQKlPtSQGNCI6ItWbFOwkfPXAXwk0FWwAbW1qYAALBgATAEkWNgNwqNiE5jYAeWZKBlIIfAA5BjjSNn19AHMjAF0lFWZOTzBgajVsACMGBjY3a1t7A31gAH5sChMAXgA+cfwACUgANTA2YjaAMg25lbW22cxxNHrlMIjo+MTk87TE7Nz8opKyitJquqCGpr4Wtrx1Lo9PoeLxDchjCbYGZzRYQXbrbBbOYAcQi8OomU4ABFqAAzMDENjAYzYA5HPxAA=

Reply:
Nice! There is a code that uses GetValueOrDefault that can be cleaned up.